python/lsst/ts/ess/common/sensor/boltek_ld250.py

```diff
@@ -0,0 +1,148 @@
+NOISE_PATTERN = re.compile(
+    rf"^\$({LD250TelemetryPrefix.NOISE_PREFIX})\*[0-9A-F]{{2}}\r\n$"
+)
+
+"""A regex pattern for a status telemetry line."""
+STATUS_PATTERN = re.compile(
+    rf"^\$({LD250TelemetryPrefix.STATUS_PREFIX}),"
+    r"(\d{1,3}),(\d{1,3}),(\d{1,2}),(\d{1,2}),(\d\d\d\.\d)\*[0-9A-F]{2}\r\n$"
+)
+
+"""A regex pattern for a strike telemetry line."""
+STRIKE_PATTERN = re.compile(
+    rf"^\$({LD250TelemetryPrefix.STRIKE_PREFIX}),(\d{{1,3}}),(\d{{1,3}}),(\d\d\d\.\d)\*[0-9A-F]{{2}}\r\n$"
+)
+
+
+class Ld250Sensor(BaseSensor):
+    """Boltek LD-250 Lightning Detector.
+
+    Perform protocol conversion for Boltek LD-250 Lightning Detector
+    instruments.
+
+    Status data is output by the instrument once per second with
+    the following format:
+
+        '$WIMST,<ccc>,<sss>,<ca>,<sa>,<hhh.h>*<cs><\r><\n>'
+
+    where:
+
+        $           Start of telemetry indicator.
+        WIMST       Telemetry prefix.
+        ccc         Close strike rate (Range 0 to 999 strikes per minute).
+        sss         Total strike rate (Range 0 to 999 strikes per minute).
+        ca          Close alarm status (0: not active, 1: active).
+        sa          Severe alarm status (0: not active, 1: active).
+        hhh.h       Current GPS heading (Range 000.0 to 359.9 deg azimuth).
+        cs          Hex checksum (Range 00 to FF).
+        <\r><\n>    2-character terminator.
+
+    Noise data is output by the instrument if the noise threshold is exceeded.
+    The threshold can be configured in the detector and is set such that the
+    noise data never, or at least as little as possible, is emitted. Still, it
+    is possible that such a message is emitted so this sensor class needs to be
+    able to process it. The format is:
+
+        '$WIMLN*<cs><\r><\n>'
+
+    where:
+
+        $           Start of telemetry indicator.
+        WIMLN       Telemetry prefix.
+        cs          Hex checksum (Range 00 to FF).
+        <\r><\n>    2-character terminator.
+
+    Strike data is output by the instrument once per minute in case one or more
+    strikes are detected. The format is:
+
+        '$WIMLI,<ddd>,<uuu>,<bbb.b>*<cs><\r><\n>'
+
+    where:
+
+        $           Start of telemetry indicator.
+        WIMLI       Telemetry prefix.
+        ddd         Corrected strike distance (Range 0 to 300 miles).
+        uuu         Uncorrected strike distance (Range 0 to 300 miles).
+        bbb.b       Bearing to the strike (Range 000.0 to 359.9 deg azimuth).
+        cs          Hex checksum (Range 00 to FF).
+        <\r><\n>    2-character terminator.
+
+    The placeholders shown for the values are displaying a non-zero-padded
+    value for int and a zero-padded value for float values.
+
+    Parameters
+    ----------
+    log : `logger`
+        The logger for which to create a child logger.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+    ) -> None:
+        super().__init__(log=log)
```

Comment:
Do you need this function of `__init__()` here?

Reply:
Thanks again. Removed as well.